fw/http_stream.c

```diff
@@ -409,34 +477,85 @@ do {									\
 						SET_STATE(new_state);
 					}
 				}
+			} else {
+				if (send)
+					ctx->cur_send_headers = stream;
+				else
+					ctx->cur_recv_headers = stream;
+			}
+			break;
+		} else if (type == HTTP2_HEADERS) {
+			TfwStream *snd_hdrs = send ? ctx->cur_send_headers
+						   : ctx->cur_recv_headers;
+
+			if (snd_hdrs) {
```

Comment:
We use TFW_H2_FSM_TYPE_CHECK early. Is it not enough?

Reply:
I believe we can rely on `TFW_H2_FSM_TYPE_CHECK` in this case.